scratchgrader/Sprite.java

```diff
@@ -63,5 +88,62 @@ public int countScripts()
             scriptsLenght.add(aux.length());
         }
         return scriptsLenght;
-    }   
+    }
+    
+    
+    /**
+     *  getAllVaraibles - Get all data variables of
+     * the sprite.
+     * @return List<String> - A list of string each string represent
+     * a data variable of the sprite.
+     */
+    public List<String> getAllVaraibles()
+    {
+        return this.dataVaraibles;
+    }
+    
+    /**
+     *  getDataVariables - Get a list that contains the 
+     * data variables of each script.
+     * @return List<String> - A list of string each string represent
+     * a data variable of a script.
+     */
+    private List<String> getDataVariables(String word)
+    {
+        List<String> dataVaraibles = new ArrayList<String>();
+        Object[] scripts = this.scripts;
+        int wordpos = 0;
+        String variable = "";
+        for(int i = 0; i < scripts.length; i++)
+        {
+            String scriptsString = scripts[i].toString();
+            wordpos = scriptsString.indexOf(word,wordpos);
+            while(wordpos != -1)
+            {
+                
+                char[] aux = scriptsString.toCharArray();
```

Comment:
I would recommend more comments  somewhere around here. It starts to get a bit confusing.